app/src/components/fields/IntegerField.tsx

```diff
@@ -0,0 +1,72 @@
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { useFormikContext } from 'formik';
+import get from 'lodash-es/get';
+import React from 'react';
+import NumberFormat, { NumberFormatProps } from 'react-number-format';
+
+export type IIntegerFieldProps = TextFieldProps & {
+  required?: boolean;
+  label: string;
+  name: string;
+  min?: number;
+  max?: number;
+};
+
+interface NumberFormatCustomProps {
+  onChange: (event: { target: { name: string; value: number } }) => void;
+  name: string;
+  min?: number;
+  max?: number;
+}
+
+const NumberFormatCustom = React.forwardRef<NumberFormatProps, NumberFormatCustomProps>(function NumericFormatCustom(
+  props,
+  ref
+) {
+  const { onChange, min = -2147483648, max = 2147483647, ...other } = props;
```

Comment:
Was gonna comment that these should use the JS max int constant, but I see these values actually align with the Postgres integer type limits. Good call.

Reply:
Thats a good point, maybe it still makes sense to use the JS max int as the default. Not really sure which use case would be more common.

Reply:
Personally I think this is a more sane default. It's already such a high number that there shouldn't really be any reason for our application to go over it and it avoids randomly raising exceptions from the DB.